staging/src/k8s.io/client-go/util/workqueue/priority_queue.go

```diff
@@ -0,0 +1,296 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package workqueue
+
+import (
+	"container/heap"
+	"sync"
+	"time"
+
+	"k8s.io/utils/clock"
+)
+
+// NewPriority constructs a new work priority queue (see the package comment).
```

Comment:
(1) There needs to be a comment defining the behavior of this type of queue.

(2) As far as I can tell from looking at the implementation here, the behavior is that items are ordered by priority *ONLY* --- the ordering of the Add calls is not relevant.  Remember that a heap does not preserve ordering between items that compare equal.  I find the irrelevance of ordering of Add among equal-priority items to be surprising.

(3) I see nothing in the package comment about priority queues.

(4) The package comment is outdated: it speaks of a single type of queue, but this package already implements multiple types of queue.